app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,20 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author = current_user
+    if @post.save
+      flash[:success] = 'Post created successfully!'
+      redirect_to user_posts_url
+    else
+      flash.now[:error] = 'Error: Post could not be created!'
+      render :new, locals: { post: @post }
+    end
+  end
```

Comment:
- [OPTIONAL] You did great using ``strong parameters`` here, but I'll suggest you extract the `post_params` into its own private function, this will make your code more modular and readable. here's a snippet of what it should look like. :+1: 

```ruby
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```